qiskit/transpiler/layout.py

```diff
@@ -393,12 +462,196 @@ class TranspileLayout:
         the circuit (and used by :meth:`.Operator.from_circuit`).
       * :attr:`final_layout` - This is a :class:`~.Layout` object used to
         model the output permutation caused ny any :class:`~.SwapGate`\s
-        inserted into the :class:~.QuantumCircuit` during the
+        inserted into the :class:`~.QuantumCircuit` during the
         :ref:`routing_stage`. It maps the output circuit's qubits from
-        :class:`.QuantumCircuit.qubits` to the final position after
-        routing.
+        :class:`.QuantumCircuit.qubits` in the output circuit to the final
+        position after routing. It is **not** a mapping from the original
+        input circuit's position to the final position at the end of the
+        transpiled circuit. If you need this you can use the
+        :meth:`.full_layout` to generate this. If this is set to ``None`` this
+        indicates that routing was not run and it can be considered equivalent
+        to a trivial layout with the qubits from the output circuit's
+        :attr:`~.QuantumCircuit.qubits` list.
```

Comment:
There's still a reference to `full_layout` here that I guess should be be `final_virtual_layout` or something?

Reply:
Fixed in: https://github.com/Qiskit/qiskit/pull/10835/commits/5acc2c01c345e7280ad7881df47fe5917d52ad97